pkg/build/graph/analysis/bc.go

```diff
@@ -64,6 +67,51 @@ bc:
 	return markers
 }
 
+// FindMissingInputImageStreams checks all build configs and confirms that their From element exists
+func FindMissingInputImageStreams(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.PredecessorNodesByEdgeKind(bcNode, buildedges.BuildInputImageEdgeKind) {
+			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
+				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+
+				if !imageStreamNode.IsFound {
+					markers = append(markers, osgraph.Marker{
+						Node:         bcNode,
+						RelatedNodes: []graph.Node{istNode},
+
+						Severity: osgraph.WarningSeverity,
+						Key:      MissingInputImageStreamWarning,
+						Message: fmt.Sprintf("%s builds from %s, but the image stream does not exist.",
+							bcNode.(*buildgraph.BuildConfigNode).ResourceString(), imageStreamNode.ResourceString()),
+					})
+
+					continue
+				}
+
+				if tagNode, ok := istNode.(*imagegraph.ImageStreamTagNode); ok {
+					imageStream := imageStreamNode.Object().(*imageapi.ImageStream)
+					if _, ok := imageStream.Status.Tags[tagNode.ImageTag()]; !ok {
```

Comment:
> I would argue that what the user has created is structurally sound and should not produce a warning. There is no additional action we'd expect the user to perform in order to have a valid build/image chain.
>  In the case where a bc does point to the istag, the action would be "do nothing"

just because a chain exists doesn't mean they will run it as a chain.  If they directly ran this buildconfig, it would sit there and do nothing, which is the situation you were trying to avoid with the original issue.